sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,808 @@
+  malloc command has completed.
+* When an asynchronous malloc command executes it will allocate USM memory of
+  the allocation kind from the memory pool.
+* When an asynchronous free command is enqueued the memory is not freed
+  immediately. The asynchronous free command may execute any time from when the
+  dependencies of the command are satisfied to when the command is synchronized
+  with. The USM pointer to be freed is no longer valid once the dependencies of
+  that command are satisfied.
+* When executing an asynchronous malloc command, a memory pool may re-allocate
+  memory from a previous USM pointer to a new USM pointer if an asynchronous
+  free command to free that memory has completed, or if there is a dependent
+  asynchronous free command enqueued such that the memory will be free by the
+  time the asynchronous malloc command executes.
+* A memory pool may reserve further memory from the memory provider at any time
+  to grow the size of the memory pool's reserve, though it may not exceed the
+  maximum size of the memory pool.
+* A memory pool may release memory back to the memory provider at any time to
+  reduce the size of the memory pool's reserve, though it may not fall below the
+  used size.
+* The amount of reserved and used memory reported by the memory pool is updated
+  immediately after returning from calls to enqueue asynchronous malloc and once
+  free commands have completed.
+* When the reserve threshold of the memory pool is set, the value specified may
+  round down to the nearest multiple of an implementation defined allocation
+  chunk size, and the memory pool will immediately start using this new size.
+* When the maximum size of the memory pool is set, the value specified may
+  round up to the nearest multiple of an implementation defined allocation chunk
+  size.
+
+Memory pools will work with both in-order and out-of-order SYCL queues.
+
+[source,c++]
+----
+namespace ext::oneapi::experimental {
+
+class memory_pool {
+
+  template <typename Properties = empty_properties_t>
+  memory_pool(const context &ctx, const device &dev, usm::alloc kind,
+    Properties props = {});
+
+  template <typename Properties = empty_properties_t>
+  memory_pool(const queue &que, usm::alloc kind, Properties props = {});
+
+  ~memory_pool();
+
+  context get_context() const;
+
+  device get_device() const;
+
+  usm::alloc get_alloc_kind() const;
+
+  size_t get_threshold() const;
+
+  size_t get_reserved_size_current() const;
+
+  size_t get_used_size_current() const;
+
+  void increase_threshold_to(size_t newThreshold);
+
+}; // memory_pool
+
+}  // ext::oneapi::experimental
+----
+
+[source, c++]
+----
+template <typename Properties = empty_properties_t>
+memory_pool(const context &ctx, const device &dev, usm::alloc kind,
+  Properties props = {});
+----
+
+_Effects_: Constructs a memory pool which is associated with the context `ctx`
+and device `dev`, which allocates either device or shared USM memory depending
```

Comment:
This sentence also mentions "shared" USM memory.

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/781c697155825026fcb412aba621277404c76901.